File: backend/app/schemas/document.py
# ============================================================
# 知识资源 schema（对齐前端 types/document.ts）
# ============================================================
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class DocumentOut(BaseModel):
    id: str
    title: str
    content: str | None = None
    folder_id: str | None = None
    folder_name: str = "收集箱"
    tags: list[str] = []
    summary: str | None = None
    ai_suggested_tags: list[str] = []
    version: int = 1
    word_count: int = 0
    created_at: datetime
    updated_at: datetime


class CreateDocumentRequest(BaseModel):
    title: str = Field(min_length=1, max_length=500)
    folder_id: str | None = None
    content: str | None = None


class UpdateDocumentRequest(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=500)
    content: str | None = None
    tags: list[str] | None = None
    folder_id: str | None = None


class DocumentVersionOut(BaseModel):
    id: str
    document_id: str
    content: str
    version: int
    word_count: int
    created_at: datetime


class BacklinkOut(BaseModel):
    id: str
    source_doc_id: str
    target_doc_id: str | None = None
    target_title: str | None = None
    created_at: datetime


class BacklinkSourceOut(BaseModel):
    source_doc_id: str
    source_title: str


class SearchTaskHit(BaseModel):
    id: str
    title: str
    type: str = "task"
    status: str = ""


class SearchDocumentHit(BaseModel):
    id: str
    title: str
    snippet: str = ""
    updated_at: datetime
    type: str = "document"


class SearchConversationHit(BaseModel):
    id: str
    title: str
    updated_at: datetime
    type: str = "conversation"


class SearchActionHit(BaseModel):
    id: str
    name: str
    type: str = "action"


class SearchResultOut(BaseModel):
    tasks: list[SearchTaskHit] = []
    documents: list[SearchDocumentHit] = []
    conversations: list[SearchConversationHit] = []
    actions: list[SearchActionHit] = []

File: frontend/src/types/document.ts
// ============================================================
// 知识资源模块类型
// 对应 PRD §13.4 知识资源模块 + §15.4-15.7 表结构
// ============================================================

export interface Folder {
  id: string
  name: string
  parent_id: string | null
  sort_order: number
  is_inbox: boolean
  created_at: string
  updated_at: string
  children?: Folder[]
}

export interface Document {
  id: string
  title: string
  /** Markdown 纯文本内容（Tiptap 编辑器序列化后存储） */
  content: string | null
  folder_id: string | null
  folder_name: string
  tags: string[]
  summary: string | null
  ai_suggested_tags: string[]
  version: number
  word_count: number
  created_at: string
  updated_at: string
}

export interface DocumentVersion {
  id: string
  document_id: string
  /** Markdown 纯文本内容 */
  content: string
  version: number
  word_count: number
  created_at: string
}

export interface Backlink {
  id: string
  source_doc_id: string
  target_doc_id: string
  target_title: string
  created_at: string
}

export interface DocumentListQuery {
  folder_id?: string
  tag?: string
  search?: string
  page?: number
  page_size?: number
  sort?: string
}

export interface CreateDocumentRequest {
  title: string
  folder_id?: string | null
  /** Markdown 纯文本内容 */
  content?: string
}

export interface UpdateDocumentRequest {
  title?: string
  /** Markdown 纯文本内容 */
  content?: string
  tags?: string[]
  folder_id?: string | null
}

export interface SearchResult {
  tasks: Array<{ id: string; title: string; type: 'task'; status: string }>
  documents: Array<{
    id: string
    title: string
    snippet: string
    updated_at: string
    type: 'document'
  }>
  conversations: Array<{
    id: string
    title: string
    updated_at: string
    type: 'conversation'
  }>
  actions: Array<{
    id: string
    name: string
    type: 'action'
  }>
}
